src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { Loader } from '../components/Loader';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types/Person';
+import { getPeopleWithParents, getPreparedPeople } from '../utils/peopleHelper';
+
+export const PeoplePage = () => {
+  const [searchParams] = useSearchParams();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setisError] = useState(false);
+  const [isDataLoaded, setIsDataLoaded] = useState(false);
+  const [hasMatches, setHasMatches] = useState(true);
+
+  const [peopleFromServer, setPeopleFromServer] = useState<Person[]>([]);
+
+  const somePeople = isDataLoaded && peopleFromServer.length !== 0;
+  const somePeopleHasMatches = somePeople && hasMatches;
+
+  const people = useMemo(
+    () => getPeopleWithParents(peopleFromServer),
+    [peopleFromServer],
+  );
+
+  const visiblePeople = getPreparedPeople(people, {
+    sex: searchParams.get('sex') || '',
+    query: searchParams.get('query') || '',
+    centuries: searchParams.getAll('centuries'),
+    sort: searchParams.get('sort') || '',
+    order: searchParams.get('order') || '',
+  });
+
+  useEffect(() => {
+    setHasMatches(visiblePeople.length > 0);
+  }, [visiblePeople]);
```

Comment:
```suggestion
  const hasMatches = visiblePeople.length > 0
```
Why it should be in state?

Reply:
I see, it's not
Thank you